### ISSUE TITLE
Screenshots are all the same - only the 1st screenshot is kept for each test

### ISSUE THREAD

**Issue by eric-grab** [square/spoon#507]
![screen shot 2017-12-20 at 5 14 53 pm](https://user-images.githubusercontent.com/31608374/34199624-57a46620-e5a9-11e7-8b5d-bf61a73a5a01.png)

My code:  
```java
@Test
    public void testInProgressTab() {
        Activity activity = activityTestRule.getActivity();
        Spoon.screenshot(activity, "InProgres-one");
        onView(withText(R.string.menu_incentives)).check(matches(isDisplayed()));
        onView(withText(R.string.in_progress)).check(matches(isDisplayed()));
        onView(withText(R.string.ended)).check(matches(isDisplayed()));
        onView(withText(R.string.active_now)).check(matches(isDisplayed()));
        onView(allOf(withId(R.id.rv_scorecards), isDisplayed())).perform(RecyclerViewActions.scrollToPosition(12));
        Spoon.screenshot(activity, "InProgres-two");// this one is not kept!
        onView(withText(R.string.everything_else)).check(matches(isDisplayed()));
    }
```

When I comment out the first `Spoon.screenshot()` then I got the 2nd screenshot correctly.

I'm using:
- Android Gradle plugin 3.0.1
- Spoon 1.7.1
- com.jaredsburrows:gradle-spoon-plugin:1.2.0

**Comment by jaredsburrows** [comment]
See this previous issue here: https://github.com/square/spoon/issues/498. Also, you need to use "single instrumentation" to get Spoon to generate the reports correctly: https://github.com/stanfy/spoon-gradle-plugin/pull/161#issuecomment-346645943.

**Comment by eric-grab** [comment]
Thanks for replying @jaredsburrows , I've already used `singleInstrumentationCall = true`. That solved 1 problem (only 1 test got screenshots ==> all tests got screenshots)
However, within 1 test like my example above, all the screenshots are the same as the first one!

**Comment by jaredsburrows** [comment]
@eric-grab Yeah, sadly I think spoon 2 has problems. Ideally, you probably only need the screenshots of the tests that have failed right?

**Comment by eric-grab** [comment]
hi @jaredsburrows 

> you probably only need the screenshots of the tests that have failed right?


serves the purpose of the debugging, yes

However, I need all the screenshots for all the steps to

- Prove that my component (a few user flows involving a few screens) sanity tests actually cover what it's supposed to cover
- Visualise the screens to help me write tests faster compared to imagining them in my mind
- Provide screenshots to product managers, marketing team etc. for as presentation materials

**Comment by jaredsburrows** [comment]
@eric-grab Makes a lot of since. I am not sure about the screenshot work around. We might have to find another solution or wait for Spoon 2 to be released. I was looking into this recently: 

Repo: https://github.com/medisafe/run-android-tests

See: https://medium.com/medisafe-tech-blog/running-android-ui-tests-part-2-15ef08056d94 and https://github.com/medisafe/run-android-tests/issues/1#issuecomment-321196601 and

**Comment by MFrat** [comment]
2020 and this problem still persists
